src/at_tools/att_supercat.rs

```diff
@@ -0,0 +1,184 @@
+        let mut corrected_paths = vec![];
+        for p in paths {
+            let candidate = match file_repair_candidates(&p, global_context.clone(), 1, true).await.get(0) {
+                Some(x) => x.clone(),
+                None => continue,
+            };
+            corrected_paths.push(candidate);
+        }
+        // drop duplicates
+        let corrected_paths = corrected_paths.into_iter().collect::<HashSet<_>>().into_iter().collect::<Vec<_>>();
+
+        let mut context_files_in = vec![];
+        
+        if !symbols_str.is_empty() {
+            let ast_arc = global_context.read().await.ast_module.clone().unwrap();
+            let ast_lock = ast_arc.read().await;
+            for p in corrected_paths.iter() {
+                
+                let mut doc = Document::new(&PathBuf::from(p));
+                doc.update_text_from_disk().await?;
+                let doc_syms = ast_lock.get_file_symbols(RequestSymbolType::All, &doc).await?
+                    .symbols.iter().map(|s|(s.name.clone(), s.guid.clone())).collect::<Vec<_>>();
+                
+                let sym_set = doc_syms.iter()
+                    .filter(|(s_name, _)| symbols_str.contains(s_name))
+                    .map(|(_, s_uuid)|s_uuid.clone())
+                    .collect::<Vec<_>>();
+                
+                let text = doc.text.map(|t|t.to_string()).unwrap_or("".to_string());
+                let cf = ContextFile {
+                    file_name: p.clone(),
+                    file_content: text.clone(),
+                    line1: 0,
+                    line2: text.lines().count(),
+                    symbol: sym_set,
+                    gradient_type: -1,
+                    usefulness: 100.0,
+                    is_body_important: false,
+                };
+                context_files_in.push(cf);
+            }
+        }
+        
+        let filenames_present = context_files_in.iter().map(|x|x.file_name.clone()).collect::<Vec<_>>();
+        for p in corrected_paths.iter().filter(|x|!filenames_present.contains(x)) {
+            let text = read_file_from_disk(&PathBuf::from(p)).await?.to_string();
+            let cf = ContextFile {
+                file_name: p.clone(),
+                file_content: text.clone(),
+                line1: 0,
+                line2: text.lines().count(),
+                symbol: vec![],
+                gradient_type: -1,
+                usefulness: 100.0,
+                is_body_important: false,
+            };
+            context_files_in.push(cf);
+        }
+        
+        let cf_out = supercat(
+            global_context.clone(),
+            tokenizer,
+            context_files_in,
+            MAX_TOKENS_PER_FILE,
+            MAX_TOKENS,
+        ).await?;
+        
+        let mut res = vec![];
+        let m = ContextEnum::ChatMessage(ChatMessage {
+            role: "supercat".to_string(),
+            content: format_context_files_to_message_content(cf_out),
```

Comment:
No, the return text should be something like 

```
file1.ext has N lines and X bytes, text below
file2.ext has N lines and X bytes, text below
```

followed by role="context_file"

Reply:
Even better if it's a tree